### PR TITLE
py-scikit-image: add 0.18.3 and fix dependencies

### PR DESCRIPTION
**Pull request by manuelakuhn**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-scikit-image/package.py [hunk 1]
```diff
@@ -13,34 +13,58 @@ class PyScikitImage(PythonPackage):
...
-    depends_on('py-pywavelets', type=('build', 'run'), when='@0.14:')
-    depends_on('py-pywavelets@1.1.1:', type=('build', 'run'), when='@0.16.1:')
+    depends_on('py-cython@0.29.21:', when='@0.18:', type='build')  # from build.txt
+    depends_on('py-cython@0.29.13:', when='@0.17:', type='build')
+    depends_on('py-cython@0.25:0.28.1,0.28.3:0.28,0.29.1:', when='@0.15:', type='build')
+    depends_on('py-cython@0.23.4:0.28.1,0.28.3:0.28,0.29.1:', when='@0.14.3:0.14', type='build')
+    depends_on('py-cython@0.23.4:0.28.1', when='@0.14.2', type='build')
+    depends_on('py-cython@0.23.4:', when='@0.14.1', type='build')
+    depends_on('py-cython@0.21:', type='build')
+    depends_on('py-numpydoc@0.6:', when='@0.13.0:0.13', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-scikit-image/package.py:41]: Do you mean `0.13.0:0.13.999` or `0.13.0`?

File: var/spack/repos/builtin/packages/py-scikit-image/package.py [hunk 2]
```diff
@@ -13,34 +13,58 @@ class PyScikitImage(PythonPackage):
...
+    depends_on('py-numpy@1.16.5:1.17,1.18.1:', when='@0.18:', type=('build', 'link', 'run'))
+    depends_on('py-numpy@1.15.1:1.17,1.18.1:', when='@0.17.0:0.17', type=('build', 'link', 'run'))
+    depends_on('py-numpy@1.14.1:', when='@0.16:', type=('build', 'link', 'run'))
+    depends_on('py-numpy@1.11:', when='@0.13:', type=('build', 'link', 'run'))
+    depends_on('py-numpy@1.7.2:', type=('build', 'run'))
+    depends_on('py-scipy@1.0.1:', when='@0.17:', type=('build', 'run'))
+    depends_on('py-scipy@0.19:', when='@0.16:', type=('build', 'run'))
+    depends_on('py-scipy@0.17:', when='@0.13:', type=('build', 'run'))
+    depends_on('py-scipy@0.9:', type=('build', 'run'))
+    depends_on('py-matplotlib@2.0:2,3.0.1:', when='@0.15:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-scikit-image/package.py:51]:
```suggestion
    depends_on('py-matplotlib@2.0:2.999,3.0.1:', when='@0.15:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-scikit-image/package.py [hunk 3]
```diff
@@ -13,34 +13,58 @@ class PyScikitImage(PythonPackage):
...
+    depends_on('py-imageio@2.3:', when='@0.16:', type=('build', 'run'))
+    depends_on('py-imageio@2.0.1:', when='@0.15:', type=('build', 'run'))
+    depends_on('py-tifffile@2019.7.26:', when='@0.17:', type=('build', 'run'))
+    depends_on('py-pywavelets@1.1.1:', when='@0.17:', type=('build', 'run'))
+    depends_on('py-pywavelets@0.4:', when='@0.13:', type=('build', 'run'))
+    depends_on('py-pooch@0.5.2:', when='@0.17.0:0.17.1', type=('build', 'run'))
+    depends_on('py-dask+array@1:', when='@0.14.2', type=('build', 'run'))
+    depends_on('py-dask+array@0.9:', when='@0.14.0:0.14.1', type=('build', 'run'))
+    depends_on('py-dask+array@0.5:', when='@:0.13', type=('build', 'run'))
+    depends_on('py-cloudpickle@0.2.1:', when='@0.14.0:0.14', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-scikit-image/package.py:70]: Same question here

**Review by adamjstewart — COMMENTED** [review]
Had a few questions about some of the version range stuff. Otherwise LGTM. Would you be interested in maintaining this package?

File: var/spack/repos/builtin/packages/py-scikit-image/package.py (hunk 2 above)
Comment (haampie) [reply var/spack/repos/builtin/packages/py-scikit-image/package.py:51]: so, `2.0:2` means `[2.0, 3)` whereas `2.0:2.999` approximates that range ;p

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**